build_wheel.sh

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+
+#************************************************#
+#           build_wheel.sh                       #
+#    Build a Python3 wheel for YOLACT            #
+#************************************************#
+
+REPO_NAME="yolact"
+NEEDED_FOLDERS=("data" "external" "layers" "scripts" "utils" "web")
+REQUIREMENTS_FILE="requirements.txt"
+SETUP_FILE="setup.py"
+PYTHON_EXTENSION=".py"
+PACKAGE_TEST_DIR="/tmp/$USER/pip_package_test"
+
+# --------------------------------------------------------- #
+# main ()                                                   #
+# drives the entire build process                           #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+main() {
+  echo "Starting build ..."
+  pre_build_setup &&
+    build &&
+    post_build_test_and_teardown &&
+    echo "Build Successful ... "
+}
+
+# --------------------------------------------------------- #
+# pre_build_setup ()                                        #
+# performs necessary setup for building a YOLACT wheel      #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+pre_build_setup() {
+  echo "Pre-Build Setup ... "
+  __validate_files && __copy_files && __fix_requirements && __build_setup
+}
+
+# --------------------------------------------------------- #
+# build ()                                                  #
+# actual build happens here                                 #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+build() {
+  echo "Build ... "
+  python3 -m build
+  rm --recursive "${REPO_NAME}"
+}
+
+# --------------------------------------------------------- #
+# post_build_test_and_teardown ()                           #
+# installation check + cleanup unnecessary files            #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+post_build_test_and_teardown() {
+  echo "Post-Build Test and Teardowns ... "
+  __test_build
+  __teardown
+}
+
+__validate_files() {
+  echo "Validating files ... "
+  # venv
+  if [[ ! -e "$(command -v python3)" ]]; then
```

Comment:
Neat, I'm not familiar with `command -v`, I usually use `which` for this behavior:
```bash
$ command -v python3
/home/danny/src/nm/.../venv/bin/python3

$ which python3
/home/danny/src/nm/.../venv/bin/python3

Reply:
I don't think I've ever used double brackets. [This article ](https://www.baeldung.com/linux/bash-single-vs-double-brackets)discusses some differences.

I believe this would achieve the same result here:
```
if [ ! -e "$(command -v python3)" ]; then
```